main/main.go

```diff
@@ -51,7 +53,9 @@ func main() {
 		os.Exit(0)
 	}
 
-	nodeConfig, err := config.GetNodeConfig(v)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	nodeConfig, cleanup, err := config.GetNodeConfig(ctx, v)
```

Comment:
ngl - the more I look at how we pass `cleanup` around, the more I think we need to do a refactor of the `config` package.

The `config` package should just be populating the config... Doing file system stuff and now networking stuff is just terrible... That being said.. It feels a bit unreasonable to expect a total refactor of node initialization just to add this one feature in... So I think this is fine for now... But we really need to clean this up because it's getting unmanageable.